pkgs/development/python-modules/python-openid/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, isPy27
+, nose
+}:
+
+buildPythonPackage rec {
+  pname = "python-openid";
+  version = "2.2.5";
+  disabled = !isPy27;
+
+  src = fetchFromGitHub {
+    owner = "openid";
+    repo = pname;
+    rev = version;
+    sha256 = "198800ka2xyjz4yi2zm23l2a0knmqzxb4wj1nhqps70jz8mwg5mz";
+  };
+
+  checkInputs = [
+    nose
+  ];
+
+  checkPhase = ''
+    nosetests
+  '';
+
+  # most tests require network access
+  doCheck = false;
+
+  meta = with lib; {
+    description = "OpenID support for servers and consumers";
+    homepage = https://github.com/openid/python-openid;
+    license = licenses.asl20;
+    maintainers = with maintainers; [ costrouc ];
```

Comment:
should probably remove costrouc as a maintainer

Reply:
> I am not sure with a way this package is packaged, since you override all version requirements, how can you be sure things will work. I did a pull request a while back that used pypi2nix, it generated a large dependency file, but at least I was sure dependencies are right.

This is a good point, I commented below the actual version we ended up choosing (based on what is available in NixOS):

```
--replace "flask>=1.1.0, <2.0" "flask" \  # got 1.1.1
--replace "flask-caching>=1.3.3, <1.4.0" "flask-caching" \ # got 1.7.2
--replace "flask-appbuilder>=1.12.5, <2.0.0" "flask-appbuilder" \ # got 2.1.6
--replace "pendulum==1.4.4" "pendulum" \ # got 2.0.5
--replace "cached_property~=1.5" "cached_property" \ # got 1.5.1
--replace "dill>=0.2.2, <0.3" "dill" \ # go 0.3.1.1
--replace "configparser>=3.5.0, <3.6.0" "configparser" \ # got 4.0.2
--replace "jinja2>=2.7.3, <=2.10.0" "jinja2" \ # got 2.10.3
--replace "funcsigs==1.0.0" "funcsigs" \ # got 1.0.2
--replace "flask-swagger==0.2.13" "flask-swagger" \ # got 0.2.14
--replace "python-daemon>=2.1.1, <2.2" "python-daemon" \ # got 2.2.3
--replace "alembic>=0.9, <1.0" "alembic" \ # got 1.2.1
--replace "markdown>=2.5.2, <3.0" "markdown" \ # 3.1.1
--replace "future>=0.16.0, <0.17" "future" \ # got 0.18.1
--replace "tenacity==4.12.0" "tenacity" \ # # got 5.1.1
--replace "text-unidecode==1.2" "text-unidecode" \ # got 1.3
--replace "tzlocal>=1.4,<2.0.0" "tzlocal" \ # got 2.0.0
--replace "sqlalchemy~=1.3" "sqlalchemy" \ # got 1.3.10
--replace "werkzeug>=0.14.1, <0.15.0" "werkzeug" # got 0.16.0
```

I do not expect markdown, sqalchemy, werkzeug or jinja2 to make wild changes on minor updates, but for those apps that made incompatible changes (like pendulum), patches to airflow have been provided. setup.py may look strict but from experience it is always a judgement call on the last person who created a release: in our case, the nixos package maintainer. After that, we only have the test suite to protect us and hopefully some module tests in the near future for integration testing.